src/cargo/ops/cargo_new.rs

```diff
@@ -541,16 +613,127 @@ fn global_config(config: &Config) -> CargoResult<CargoNewConfig> {
         }
         None => None
     };
+
     Ok(CargoNewConfig {
         name: name,
         email: email,
         version_control: vcs,
     })
 }
 
+/// Recursively list directory contents under `dir`, only visiting files.
+///
+/// This will also filter out files & files types which we don't want to
+/// try generate templates for. Image files, for instance.
+///
+/// It also filters out certain files & file types, as we don't want t
+///
+/// We use this instead of std::fs::walk_dir as it is marked as unstable for now
+///
+/// This is a modified version of the example at:
+///    http://doc.rust-lang.org/std/fs/fn.read_dir.html
+fn walk_template_dir(dir: &Path, cb: &mut FnMut(DirEntry) -> CargoResult<()>) -> CargoResult<()> {
+    let attr = try!(fs::metadata(&dir));
+    let ignore_files = vec![".gitignore"];
+
+    if attr.is_dir() {
+        for entry in try!(fs::read_dir(dir)) {
```

Comment:
You can cut down on the indentation here with an early return of `if !attr.is_dir()` perhaps?